src/cargo/ops/cargo_update.rs

```diff
@@ -514,12 +514,14 @@ fn print_lockfile_generation(
         };
 
         for package in diff.added.iter() {
+            let required_rust_version = report_required_rust_version(ws, resolve, *package);
             let latest = report_latest(&possibilities, *package);
+            let note = required_rust_version.or(latest);
```

Comment:
nit: should we make `report_latest` lazy via `or_else`?
(I know we keep this for consistency for the unchanged branch, though)

Reply:
The consistency with `unchanged` was my motivation with the assumption that this wouldn't have a meaningful performance impact.  Consistency has the benefit of making it easier to spot branches you need to update (or that a branch was forgotten)